Source/creature_internal.cpp

```diff
@@ -79,13 +80,19 @@ TOperator* creature_internal::_get_genetic_operator(TMetadata** operator_metadat
 }
 
 crossover* creature_internal::_get_crossover(crossover_metadata** metadata, size_t num_crossovers) {
-    return _get_genetic_operator<crossover, crossover_metadata>(metadata, num_crossovers,
-            create_genetic_operator_a<crossover, crossover_metadata>(genetic_factory::create_crossover));
+    genetic_factory& factory = genetic_factory::get_instance();
```

Comment:
Why do we need to create a `genetic_factory&` object? Why, not a `genetic_factory*`?

Reply:
genetic_factory& is a 'reference' to a genetic_factory object while genetic_factory* is a 'pointer' to a genetic_factory object. They behave similarly but they are not semantically the same. In this case if we where to use a pointer we would have to write code to delete the global genetic factory pointer at the end of a galapagos session. If on the other hand we where to use a reference the genetic_factory memory would automatically get released when the static portion of genetic_factory goes out of scope. For more information on the distinction between a pointer and a reference type see https://stackoverflow.com/questions/57483/what-are-the-differences-between-a-pointer-variable-and-a-reference-variable-in